packages/sanity/src/tasks/src/tasks/components/activity/TasksActivityLog.tsx

```diff
@@ -0,0 +1,250 @@
+  threadCommentsAvatar: true,
+  replyAvatar: true,
+  avatarSize: 0,
+}
+
+const MotionStack = styled(motion(Stack))``
+
+interface TasksActivityLogProps {
+  onChange: (patch: FormPatch | PatchEvent | FormPatch[]) => void
+  path?: Path
+  value: TaskDocument
+}
+
+interface ActivityLogItem {
+  author: string
+  field: string
+  from: string
+  timestamp: string
+  to?: string
+}
+
+type Activity =
+  | {
+      _type: 'comment'
+      payload: CommentThreadItem
+      timestamp: string
+    }
+  | {
+      _type: 'activity'
+      payload: ActivityLogItem
+      timestamp: string
+    }
+
+export function TasksActivityLog(props: TasksActivityLogProps) {
+  const {value, onChange, path} = props
+  const currentUser = useCurrentUser()
+
+  const {comments, mentionOptions, operation, getComment} = useComments()
+  const loading = comments.loading
+  const taskComments = comments.data.open
+
+  const handleCommentCreate = useCallback(
+    (message: CommentInputProps['value']) => {
+      const nextComment: CommentCreatePayload = {
+        type: 'task',
+        message,
+        parentCommentId: undefined,
+        reactions: EMPTY_ARRAY,
+        status: 'open',
+        threadId: uuid(),
+      }
+
+      operation.create(nextComment)
+    },
+    [operation],
+  )
+
+  const handleCommentReply = useCallback(
+    (nextComment: CommentBaseCreatePayload) => {
+      operation.create({
+        type: 'task',
+        message: nextComment.message,
+        parentCommentId: nextComment.parentCommentId,
+        reactions: EMPTY_ARRAY,
+        status: 'open',
+        threadId: nextComment.threadId,
+      })
+    },
+    [operation],
+  )
+
+  const handleCommentCreateRetry = useCallback(
+    (id: string) => {
+      // Get the optimistically created comment and use it
+      // when retrying the creation.
+      const comment = getComment(id)
+      if (!comment) return
+
+      operation.create({
+        type: 'task',
+        id: comment._id,
+        message: comment.message,
+        parentCommentId: comment.parentCommentId,
+        reactions: comment.reactions || EMPTY_ARRAY,
+        status: comment.status,
+        threadId: comment.threadId,
+      })
+    },
+    [getComment, operation],
+  )
+
+  const handleCommentReact = useCallback(
+    (id: string, reaction: CommentReactionOption) => {
+      operation.react(id, reaction)
+    },
+    [operation],
+  )
+
+  const handleCommentRemove = useCallback(
+    (id: string) => {
+      // TODO:
+      // The remove operation is not optimistic. We should display a
+      // dialog to confirm the removal and wait for the server to respond
+      // before removing the comment from the UI. (See `CommentsDocumentInspector`)
+      operation.remove(id)
+    },
+    [operation],
+  )
+
+  const handleCommentEdit = useCallback(
+    (id: string, next: CommentUpdatePayload) => {
+      operation.update(id, next)
+    },
+    [operation],
+  )
+
+  // TODO: Get the task real activity.
```

Comment:
This is coming as part of https://github.com/sanity-io/sanity/tree/feat/edx-1137